src/Extensions/Hosting.Services.Web/ApplicationBuilderExtensions.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Net;
+using System.Text.Json;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Diagnostics;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Omex.Extensions.Hosting.Services.Web.Middlewares;
+
+namespace Microsoft.Omex.Extensions.Hosting.Services.Web
+{
+	/// <summary>
+	/// Extension to add Omex dependencies to ApplicationBuilder
+	/// </summary>
+	public static class ApplicationBuilderExtensions
+	{
+		/// <summary>
+		/// Adds the default exception heandling logic that will display developer expeption page in develop and short message during deployment
+		/// </summary>
+		/// <param name="builder">Application builder</param>
+		/// <param name="acceptObsoleteCorrelationHeader">Set it to true if service need to accept request with old style correlation</param>
+		public static IApplicationBuilder UseOmexMiddlewares(this IApplicationBuilder builder, bool acceptObsoleteCorrelationHeader)
```

Comment:
>acceptObsoleteCorrelationHeader [](start = 94, length = 31)

I am thinking, would the name enableBackwardCompatibility better in this context?

Reply:
I agree that it sounds much better, but BackwardCompatibility seems a bit too generic to me because there might be many compatibility scenarios.

Reply:
I think it's fine either way, but an more specific alternative for Daniel's suggestion could be: enableCorrelationHeaderCompatibility or enableCorrelationHeaderBackwardCompatibility.

Reply:
I like `enableCorrelationHeaderBackwardCompatibility`, thank you